### PR TITLE
First version of a pickup magnet

### PR DESCRIPTION
**Pull request by Kevin22888**
- A point effector 2D and a circle collider, with some parameter tweaking.
- Fixed a small bug where pickups stick to the walls (due to a function that is reducing movement, which is now removed)

Interesting Note:
When a pickup is starting to get attracted towards the player, if we quickly run/jump away, the pickup will be left behind.
This may not necessary make the magnet that effective, so we could consider some more changes.

### CODE REVIEW

**Comment by eon704** [comment]
Magnet feels nice!

A little comment:
- At times it feels that the magnet's range needs to be longer - how does it feel if we increase the magnet range by 50%?

**Comment by Kevin22888** [comment]
Changes:
- Removed Point Effector and extra collider and added a custom script for the magnet effect

`ResourceMagnet` script on the player:
- can customize the attraction radius, and min/max speed
- use Lerp to set attraction speed based on min/max settings

File: Assets/Code/Scripts/Core/Player/ResourceMagnet.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                if (hitColliders[i] == null) continue;
+
+                ResourcePickup pickup = hitColliders[i].GetComponent<ResourcePickup>();
+                if (pickup != null)
+                {
+                    Vector2 directionToPlayer = (Vector2)(transform.position - pickup.transform.position);
+                    float distanceToPlayer = directionToPlayer.magnitude;
+                    
+                    // Choose a speed based on the pickup's distance to the player
+                    float speed = Mathf.Lerp(_minSpeed, _maxSpeed, distanceToPlayer / _attractionRadius);
```
Comment (Eryngii01) [inline Assets/Code/Scripts/Core/Player/ResourceMagnet.cs:56]: I believe that when the resource pickup is farthest from the player's attraction radius, the `distanceToPlayer` will be maximized and will begin moving at the `_maxSpeed` which will eventually lerp to the `_minSpeed` as the pickup moves closer. 

If this was intended, feel free to disregard this comment 👍

Comment (eon704) [reply Assets/Code/Scripts/Core/Player/ResourceMagnet.cs:56]: Personally, it appears visually better to my eye as it is, starting at the max speed and then gradually slowing down 🤔

**Review by eon704 — APPROVED** [review]
Overall looks good!

Personally, I think the value of 40 for the `MaxSpeed` is a bit too much. I would put it somewhere in the range of 20-25. And as for the `MinSpeed` I think having it around 12-14 appears quite nice. But these are personal preferences, if you would like to change those to something else feel free to do so 👌

**Pull request merged**